cmd/libs/go2idl/conversion-gen/generators/conversion.go

```diff
@@ -526,26 +556,34 @@ func (g *genConversion) doStruct(inType, outType *types.Type, sw *generator.Snip
 			"outType": outMember.Type,
 			"name":    m.Name,
 		}
+		if function, ok := g.preexists(m.Type, outMember.Type); ok {
+			args["function"] = function
+			sw.Do("if err := $.function|raw$(&in.$.name$, &out.$.name$, s); err != nil {\n", args)
+			sw.Do("return err\n", nil)
+			sw.Do("}\n", nil)
+			continue
+		}
 		switch m.Type.Kind {
 		case types.Builtin:
-			sw.Do("out.$.name$ = in.$.name$\n", args)
+			if m.Type == outMember.Type {
+				sw.Do("out.$.name$ = in.$.name$\n", args)
+			} else {
+				sw.Do("out.$.name$ = $.outType|raw$(in.$.name$)\n", args)
+			}
 		case types.Map, types.Slice, types.Pointer:
 			sw.Do("if in.$.name$ != nil {\n", args)
-			sw.Do("in, out := in.$.name$, &out.$.name$\n", args)
+			sw.Do("in, out := in.$.name$, out.$.name$\n", args)
 			g.generateFor(m.Type, outMember.Type, sw)
 			sw.Do("} else {\n", nil)
 			sw.Do("out.$.name$ = nil\n", args)
 			sw.Do("}\n", nil)
 		case types.Struct:
-			if function, ok := g.preexists(m.Type, outMember.Type); ok {
-				args["function"] = function
-				sw.Do("if err := $.function|raw$(in.$.name$, &out.$.name$, s); err != nil {\n", args)
-			} else if g.convertibleOnlyWithinPackage(m.Type, outMember.Type) {
+			if g.convertibleOnlyWithinPackage(m.Type, outMember.Type) {
 				funcName := g.funcNameTmpl(m.Type, outMember.Type)
-				sw.Do(fmt.Sprintf("if err := %s(in.$.name$, &out.$.name$, c); err != nil {\n", funcName), args)
+				sw.Do(fmt.Sprintf("if err := %s(&in.$.name$, &out.$.name$, s); err != nil {\n", funcName), args)
```

Comment:
Yeah - that sounds good. Although, since I already did that, I will just add a TODO to fix that in the future :)